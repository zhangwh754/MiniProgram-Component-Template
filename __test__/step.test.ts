const path = require('path')
const simulate = require('miniprogram-simulate')

describe('MStepProgress', () => {
    const Step = simulate.load(path.join(process.cwd(), './dist/components/MStepProgress/index'))

    test('renders basic step progress with default props', async () => {
        const wrapper = simulate.render(simulate.load({
            usingComponents: { 'MStepProgress': Step },
            template: `<MStepProgress steps="{{steps}}" />`,
            data: {
                steps: [
                    { label: 'Step 1', timestamp: '2024-03-20 10:00' },
                    { label: 'Step 2', timestamp: '2024-03-21 10:00' },
                    { label: 'Step 3', timestamp: '2024-03-22 10:00' }
                ]
            }
        }))

        wrapper.attach(document.createElement('parent-wrapper'))
        expect(wrapper.toJSON()).toMatchSnapshot()
    })

    test('renders with custom colors', async () => {
        const wrapper = simulate.render(simulate.load({
            usingComponents: { 'MStepProgress': Step },
            template: `<MStepProgress 
                steps="{{steps}}" 
                primaryColor="{{primaryColor}}"
                secondColor="{{secondColor}}"
            />`,
            data: {
                steps: [
                    { label: 'Step 1', timestamp: '2024-03-20 10:00' },
                    { label: 'Step 2', timestamp: '2024-03-21 10:00' }
                ],
                primaryColor: '#ff0000',
                secondColor: '#dddddd'
            }
        }))

        wrapper.attach(document.createElement('parent-wrapper'))
        expect(wrapper.toJSON()).toMatchSnapshot()
    })

    test('renders with active index and useIndex', async () => {
        const wrapper = simulate.render(simulate.load({
            usingComponents: { 'MStepProgress': Step },
            template: `<MStepProgress 
                steps="{{steps}}" 
                activeIndex="{{activeIndex}}"
                useIndex="{{useIndex}}"
            />`,
            data: {
                steps: [
                    { label: 'Step 1', timestamp: '2024-03-20 10:00' },
                    { label: 'Step 2', timestamp: '2024-03-21 10:00' },
                    { label: 'Step 3', timestamp: '2024-03-22 10:00' }
                ],
                activeIndex: 1,
                useIndex: true
            }
        }))

        wrapper.attach(document.createElement('parent-wrapper'))
        expect(wrapper.toJSON()).toMatchSnapshot()
    })
})