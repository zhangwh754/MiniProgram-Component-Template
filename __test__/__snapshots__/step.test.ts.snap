// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MStepProgress renders basic step progress with default props 1`] = `
<main>
  <main>
    <wx-view
      class="m-step-progress horizontal"
      style="--primary-color: #1677FF;--second-color: #ccc"
    >
      <wx-view
        class="step-item completed"
        data-index="{{0}}"
        data-item="{{
          {
            "label": "Step 1",
            "timestamp": "2024-03-20 10:00",
          }
        }}"
        bind:tap="onStepItemClick"
      >
        <wx-view
          class="step-circle"
        >
          <wx-view
            class="icon-check"
          >
            ✓
          </wx-view>
        </wx-view>
        <wx-view
          class="step-content"
        >
          <wx-view
            class="step-label"
          >
            Step 1
          </wx-view>
          <wx-view
            class="step-timestamp"
          >
            2024-03-20 10:00
          </wx-view>
        </wx-view>
        <wx-view
          class="step-connector"
        />
      </wx-view>
      <wx-view
        class="step-item "
        data-index="{{1}}"
        data-item="{{
          {
            "label": "Step 2",
            "timestamp": "2024-03-21 10:00",
          }
        }}"
        bind:tap="onStepItemClick"
      >
        <wx-view
          class="step-circle"
        >
          <wx-view
            class="step-number"
          >
            2
          </wx-view>
        </wx-view>
        <wx-view
          class="step-content"
        >
          <wx-view
            class="step-label"
          >
            Step 2
          </wx-view>
          <wx-view
            class="step-timestamp"
          >
            2024-03-21 10:00
          </wx-view>
        </wx-view>
        <wx-view
          class="step-connector"
        />
      </wx-view>
      <wx-view
        class="step-item "
        data-index="{{2}}"
        data-item="{{
          {
            "label": "Step 3",
            "timestamp": "2024-03-22 10:00",
          }
        }}"
        bind:tap="onStepItemClick"
      >
        <wx-view
          class="step-circle"
        >
          <wx-view
            class="step-number"
          >
            3
          </wx-view>
        </wx-view>
        <wx-view
          class="step-content"
        >
          <wx-view
            class="step-label"
          >
            Step 3
          </wx-view>
          <wx-view
            class="step-timestamp"
          >
            2024-03-22 10:00
          </wx-view>
        </wx-view>
        <wx-view
          class="step-connector"
        />
      </wx-view>
    </wx-view>
  </main>
</main>
`;

exports[`MStepProgress renders with active index and useIndex 1`] = `
<main>
  <main>
    <wx-view
      class="m-step-progress horizontal"
      style="--primary-color: #1677FF;--second-color: #ccc"
    >
      <wx-view
        class="step-item completed"
        data-index="{{0}}"
        data-item="{{
          {
            "label": "Step 1",
            "timestamp": "2024-03-20 10:00",
          }
        }}"
        bind:tap="onStepItemClick"
      >
        <wx-view
          class="step-circle"
        >
          <wx-view
            class="step-number"
          >
            1
          </wx-view>
        </wx-view>
        <wx-view
          class="step-content"
        >
          <wx-view
            class="step-label"
          >
            Step 1
          </wx-view>
          <wx-view
            class="step-timestamp"
          >
            2024-03-20 10:00
          </wx-view>
        </wx-view>
        <wx-view
          class="step-connector"
        />
      </wx-view>
      <wx-view
        class="step-item completed"
        data-index="{{1}}"
        data-item="{{
          {
            "label": "Step 2",
            "timestamp": "2024-03-21 10:00",
          }
        }}"
        bind:tap="onStepItemClick"
      >
        <wx-view
          class="step-circle"
        >
          <wx-view
            class="step-number"
          >
            2
          </wx-view>
        </wx-view>
        <wx-view
          class="step-content"
        >
          <wx-view
            class="step-label"
          >
            Step 2
          </wx-view>
          <wx-view
            class="step-timestamp"
          >
            2024-03-21 10:00
          </wx-view>
        </wx-view>
        <wx-view
          class="step-connector"
        />
      </wx-view>
      <wx-view
        class="step-item "
        data-index="{{2}}"
        data-item="{{
          {
            "label": "Step 3",
            "timestamp": "2024-03-22 10:00",
          }
        }}"
        bind:tap="onStepItemClick"
      >
        <wx-view
          class="step-circle"
        >
          <wx-view
            class="step-number"
          >
            3
          </wx-view>
        </wx-view>
        <wx-view
          class="step-content"
        >
          <wx-view
            class="step-label"
          >
            Step 3
          </wx-view>
          <wx-view
            class="step-timestamp"
          >
            2024-03-22 10:00
          </wx-view>
        </wx-view>
        <wx-view
          class="step-connector"
        />
      </wx-view>
    </wx-view>
  </main>
</main>
`;

exports[`MStepProgress renders with custom colors 1`] = `
<main>
  <main>
    <wx-view
      class="m-step-progress horizontal"
      style="--primary-color: #ff0000;--second-color: #dddddd"
    >
      <wx-view
        class="step-item completed"
        data-index="{{0}}"
        data-item="{{
          {
            "label": "Step 1",
            "timestamp": "2024-03-20 10:00",
          }
        }}"
        bind:tap="onStepItemClick"
      >
        <wx-view
          class="step-circle"
        >
          <wx-view
            class="icon-check"
          >
            ✓
          </wx-view>
        </wx-view>
        <wx-view
          class="step-content"
        >
          <wx-view
            class="step-label"
          >
            Step 1
          </wx-view>
          <wx-view
            class="step-timestamp"
          >
            2024-03-20 10:00
          </wx-view>
        </wx-view>
        <wx-view
          class="step-connector"
        />
      </wx-view>
      <wx-view
        class="step-item "
        data-index="{{1}}"
        data-item="{{
          {
            "label": "Step 2",
            "timestamp": "2024-03-21 10:00",
          }
        }}"
        bind:tap="onStepItemClick"
      >
        <wx-view
          class="step-circle"
        >
          <wx-view
            class="step-number"
          >
            2
          </wx-view>
        </wx-view>
        <wx-view
          class="step-content"
        >
          <wx-view
            class="step-label"
          >
            Step 2
          </wx-view>
          <wx-view
            class="step-timestamp"
          >
            2024-03-21 10:00
          </wx-view>
        </wx-view>
        <wx-view
          class="step-connector"
        />
      </wx-view>
    </wx-view>
  </main>
</main>
`;
